### PR TITLE
47 validate config files

### PR DESCRIPTION
**Pull request by forbidden-arts**
Hello Stranger still works
shifted from shared_ptr to a normal vector<server> 
cleaned up main. tested the bejezzus out of this, but please feel free to kick the tires.

### CODE REVIEW

**Review by rosamakinen — APPROVED** [review]
looks good to me!

File: src/ConfigParser.cpp [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
-	for (vectorMap::iterator it = vMap.begin(); it != vMap.end(); ++it) {
-		it->second.clear();
-	}
-	vMap.clear();
+}
+
+static void configError(const std::string& str, size_t lineNumber)
+{
+	std::string base = "Config File Error: ";
+	base += str + " Line: " + std::to_string(lineNumber);
```
Comment (merituulie) [inline src/ConfigParser.cpp:15]: I think itll be safer to use the string function append, concatination is usually quite expensive and overloadable. Just a suggestion, though I've been changing these in other places too!

File: src/ConfigParser.cpp [hunk 2]
```diff
@@ -18,21 +25,142 @@ static std::string trim(const std::string& str)
...
+}
+
 void ConfigParser::parseConfig(const std::string& filename)
 {
 	std::ifstream file(filename);
 	std::string line;
 
 	if (!file.is_open())
-		throw ConfigurationException("Failed to open file.");
+		configError("Failed to open file.", lineNumber);
```
Comment (merituulie) [inline src/ConfigParser.cpp:156]: I added some checks for the FileHandler here, .bad() and another one, maybe those could be added here too? Also, why the lineNumber here now that we just cant open the file?

File: src/ConfigParser.cpp [hunk 3]
```diff
@@ -48,63 +176,55 @@ void ConfigParser::parseConfig(const std::string& filename)
 					else
 						currentSection.clear();
 				}
```
Comment (merituulie) [inline src/ConfigParser.cpp:178]: Just a general note, there is quite a lot of nesting here! we could just check if the sectionstack is empty and throw in that case, where we would not have to have so many brackets etc! no need for a change, small thing and more for longer lasting projects but just noting!

File: src/main.cpp [hunk 4]
```diff
@@ -48,12 +41,11 @@ int main(int argc, char **argv)
 		servers.push_back(server);
 		serverHandler.runServers(servers);
 
-		servers.clear();
+			servers.clear();
```
Comment (merituulie) [inline src/main.cpp:44]: whats with these tabs?

File: src/main.cpp [hunk 5]
```diff
@@ -33,8 +18,16 @@ Server& initServer()
...
+	{	
+		try
+		{
+			std::cout << "This is a placeholder for the default config.";
+			return 0;
+		}
+		catch(const std::logic_error& e)
+		{
+			std::cerr << e.what() << '\n';
+		}
```
Comment (merituulie) [inline src/main.cpp:30]: I dont really get whats going on here, are we not going to add this parsing into the same try catch as the rest? this also now just exits straight up if we dont provide an argument as input. Could just be taken out and just cerr "Expecting at least one input" or smt..

**Pull request merged**